src/dispatching/dispatcher.rs

```diff
@@ -561,8 +562,8 @@ impl std::error::Error for IdleShutdownError {}
 /// A token which used to shutdown [`Dispatcher`].
 #[derive(Clone)]
 pub struct ShutdownToken {
-    dispatcher_state: Arc<DispatcherState>,
-    shutdown_notify_back: Arc<Notify>,
+    pub(crate) dispatcher_state: Arc<DispatcherState>,
+    pub(crate) shutdown_notify_back: Arc<Notify>,
 }
```

Comment:
The fact that this needs so many `pub(crate)` is a bad sign. I'll try to move it to a better place later, in a separate PR.